mfr/extensions/jasp/html_processor.py

```diff
@@ -0,0 +1,63 @@
+
+from io import StringIO
+from html.parser import HTMLParser
+
+import base64
+
+class HTMLProcessor(HTMLParser):
+
+    # The HTMLProcessor replaces the src attribute in <image> tags with the base64 equivalent
+    # The image content comes from the zip_file (specified with set_src_source())
+    # It also strips <script> and <object> tags from the HTML (potential attack vectors)
+
+    html = StringIO()  # buffer for the processed HTML
```

Comment:
Attempting to render the same file (probably also different files, but I only have `ok.jasp` and `contains-malicious-script.jasp` to work with) just appends the new data to the old render because this `html` buffer is never cleared.

<img width="1280" alt="once" src="https://cloud.githubusercontent.com/assets/5659262/9969042/0d303966-5e1a-11e5-9011-100a09909a14.png">
<img width="1280" alt="twice" src="https://cloud.githubusercontent.com/assets/5659262/9969047/0f375d34-5e1a-11e5-9b1c-8eef9802a604.png">
<img width="1280" alt="many times" src="https://cloud.githubusercontent.com/assets/5659262/9969049/113af94c-5e1a-11e5-84b4-b2808ff6aea3.png">